File: Java/src/com/stephenlhc/RemoveNthNodeFromEndofList.java
package com.stephenlhc;

public class RemoveNthNodeFromEndofList {
//  19. Remove Nth Node From End of List

    public ListNode removeNthFromEnd(ListNode head, int n) {
        var slow = head;
        var fast = head;

        for(int i = 0; i < n; i++)
            fast = fast.next;

        if(fast == null){
            head = head.next;
            return head;
        }

        while(fast.next != null){
            slow = slow.next;
            fast = fast.next;
        }

        slow.next = slow.next.next;

        return head;
    }
}

File: Python/Remove_Nth_Node_From_End_of_List.py
# 19. Remove Nth Node From End of List

def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
    fast = head
    slow = head
    for _ in range(n):
        fast = fast.next

    if not fast:
        return head.next

    while fast.next:
        fast = fast.next
        slow = slow.next
    slow.next = slow.next.next
    return head
